### PR TITLE
[Update] change mysql-client to mycli

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-connection-parameters.md [hunk 1]
```diff
@@ -241,7 +241,7 @@ UPDATE `t` SET `a` = 11 WHERE `id` = 2;
 UPDATE `t` SET `a` = 12 WHERE `id` = 3;
 ```
 
-另外，因为一个[客户端 bug](https://bugs.mysql.com/bug.php?id=96623)，批量更新时如果要配置 `rewriteBatchedStatements = true` 和 `useServerPrepStmts = true`，推荐同时配置 `allowMultiQueries = true` 参数来避免这个 bug。
+另外，因为一个[MySQL JDBC bug](https://bugs.mysql.com/bug.php?id=96623)，批量更新时如果要配置 `rewriteBatchedStatements = true` 和 `useServerPrepStmts = true`，推荐同时配置 `allowMultiQueries = true` 参数来避免这个 bug。
```
Comment (Oreoxmt) [inline develop/dev-guide-connection-parameters.md:244]:
```suggestion
另外，因为一个 [MySQL JDBC bug](https://bugs.mysql.com/bug.php?id=96623)，批量更新时如果要配置 `rewriteBatchedStatements = true` 和 `useServerPrepStmts = true`，推荐同时配置 `allowMultiQueries = true` 参数来避免这个 bug。
```

File: develop/dev-guide-sample-application-golang.md [hunk 2]
```diff
@@ -822,7 +822,7 @@ dsn := "root:123456@tcp(xxx.tidbcloud.com:4000)/test?charset=utf8mb4"
 
 运行 `make all`，这是以下三个操作的组合：
 
-- 创建表 (make mysql)：`mysql --host 127.0.0.1 --port 4000 -u root<sql/dbinit.sql`
+- 创建表 (make prepare)：`mycli --host 127.0.0.1 --port 4000 -u root --no-warn < sql/dbinit.sql`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:825]:
```suggestion
- 创建表 (`make prepare`)：`mycli --host 127.0.0.1 --port 4000 -u root --no-warn < sql/dbinit.sql`
```

File: develop/dev-guide-sample-application-golang.md [hunk 3]
```diff
@@ -822,7 +822,7 @@ dsn := "root:123456@tcp(xxx.tidbcloud.com:4000)/test?charset=utf8mb4"
...
 - 构建二进制 (make build)： `go build -o bin/sql-driver-example`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:826]:
```suggestion
- 构建二进制 (`make build`)： `go build -o bin/sql-driver-example`
```

File: develop/dev-guide-sample-application-golang.md [hunk 4]
```diff
@@ -822,7 +822,7 @@ dsn := "root:123456@tcp(xxx.tidbcloud.com:4000)/test?charset=utf8mb4"
...
 - 运行 (make run)： `./bin/sql-driver-example`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-golang.md:827]:
```suggestion
- 运行 (`make run`)：`./bin/sql-driver-example`
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 5]
```diff
@@ -30,93 +30,60 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
 
 ## 第 2 步：连接到集群
 
-1. 若未安装 MySQL 客户端，请选择自己的操作系统，按以下步骤安装。
+1. 若未安装 MySQL 客户端，请选择自己的操作系统，按以下步骤安装。推荐使用 [mycli](https://www.mycli.net/)，这是一个使用 Python 编写的 MySQL 客户端，其比 MySQL 官方的 mysql-client 更加智能和现代化。
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:33]:
```suggestion
1. 若未安装 MySQL 客户端，请根据你的操作系统，按以下步骤安装。推荐使用 [mycli](https://www.mycli.net/)，这是一个使用 Python 编写的、对开发者友好的现代化 MySQL 客户端。
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 6]
```diff
@@ -30,93 +30,60 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
...
 
     <SimpleTab>
 
     <div label="macOS">
 
-    如果你没有安装 Homebrew，请移步 [Homebrew 官网](https://brew.sh/index_zh-cn)进行安装。
+    如果你没有安装 Homebrew，请移步 [Homebrew 官网](https://brew.sh/index_zh-cn)进行安装。安装后等待安装完毕即可：
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:39]:
```suggestion
    使用 Hombrew 安装 `mycli`。如果你没有安装 Homebrew，请参考 [Homebrew 官网](https://brew.sh/index_zh-cn)进行安装。
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 7]
```diff
@@ -30,93 +30,60 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
...
-    mysql  Ver 15.1 Distrib 5.5.68-MariaDB, for Linux (x86_64) using readline 5.1
+    mysql --connect-timeout 15 -u <user> -h <host> -P <port> -p
     ```
 
-    </div>
-
-    </SimpleTab>
-
-2. 运行第 1 步中得到的连接字符串。
+    则需改为运行：
```
Comment (Oreoxmt) [inline develop/dev-guide-build-cluster-in-cloud.md:81]:
```suggestion
    则需修改为：
```

File: develop/dev-guide-connect-to-tidb.md [hunk 8]
```diff
@@ -6,22 +6,18 @@ aliases: ['/zh/tidb/dev/connect-to-tidb']
...
-**TiDB** 高度兼容 **MySQL 5.7** 协议，全量的客户端链接参数列表，请参阅 [MySQL Client Options](https://dev.mysql.com/doc/refman/5.7/en/mysql-command-options.html)。
+**TiDB** 高度兼容 **MySQL 5.7** 协议，请参阅 [MySQL 客户端/服务器协议](https://dev.mysql.com/doc/internals/en/client-server-protocol.html)。这使得大多数客户端驱动程序和 ORM 框架可以像连接到 MySQL 一样地连接到 TiDB。
 
-TiDB 支持 [MySQL 客户端/服务器协议](https://dev.mysql.com/doc/internals/en/client-server-protocol.html)。这使得大多数客户端驱动程序和 ORM 框架可以像连接到 MySQL 一样地连接到 TiDB。
+## TiDB 命令行工具
 
-## MySQL Client
-
-你可以使用 MySQL Client 作为 TiDB 的命令行工具。在 [MySQL Shell 官方文档](https://dev.mysql.com/doc/mysql-shell/8.0/en/mysql-shell-install.html) 你可以找到不同操作系统的安装方式。在安装完后你可以使用如下命令行连接到 TiDB：
+PingCAP 推荐使用 [mycli](https://www.mycli.net/) 作为 TiDB 的命令行工具，这是一个使用 Python 编写的 MySQL 客户端，其比 MySQL 官方的 mysql-client 更加智能和现代化。在 [mycli 官方文档](https://www.mycli.net/install) 你可以找到不同操作系统的安装方式。在安装完后你可以使用如下命令行连接到 TiDB：
```
Comment (Oreoxmt) [inline develop/dev-guide-connect-to-tidb.md:13]:
```suggestion
推荐使用 [mycli](https://www.mycli.net/) 作为 TiDB 的命令行工具，这是一个使用 Python 编写的 MySQL 客户端、对开发者友好的现代化 MySQL 客户端。不同操作系统的 mycli 安装方式请参考 [mycli 官网](https://www.mycli.net/install)。安装完成后，你可以使用如下命令连接到 TiDB：
```

File: develop/dev-guide-create-table.md [hunk 9]
```diff
@@ -412,7 +413,7 @@ SHOW TABLES IN `bookshop`;
 
 ### `CREATE TABLE` 执行时应遵守的规则
 
-- 不推荐使用客户端的 Driver 或 ORM 来执行数据库模式的更改。基于过往经验，建议使用 [MySQL 客户端](https://dev.mysql.com/doc/refman/8.0/en/mysql.html)或使用任意你喜欢的 GUI 客户端来进行数据库模式的更改。本文档中，将在大多数场景下，使用 **MySQL 客户端** 传入 SQL 文件来执行数据库模式的更改。
+- 不推荐使用客户端的 Driver 或 ORM 来执行数据库模式的更改。基于过往经验，建议使用 [mycli 客户端](https://www.mycli.net/)，这是一个使用 Python 编写的 MySQL 客户端，其比 MySQL 官方的 mysql-client 更加智能和现代化。或使用任意你喜欢的 GUI 客户端来进行数据库模式的更改。本文档中，将在大多数场景下，使用 **mycli 客户端** 传入 SQL 语句或 SQL 文件来执行数据库模式的更改。
```
Comment (Oreoxmt) [inline develop/dev-guide-create-table.md:416]:
```suggestion
- 不推荐使用客户端的 Driver 或 ORM 来执行数据库模式的更改。基于过往经验，建议使用 [mycli 客户端](https://www.mycli.net/)，这是一个使用 Python 编写的、对开发者友好的现代化 MySQL 客户端。或使用任意你喜欢的 GUI 客户端来进行数据库模式的更改。本文档的大多数场景下，使用 **mycli 客户端** 传入 SQL 语句或 SQL 文件来执行数据库模式的更改。
```

**Comment by Icemap** [comment]
不直接替换 `mysql-client`。而是使用 Tips 推荐 `mycli`。

**Pull request closed without merging**